### PR TITLE
[ready]Validate if token replacer is global or per index

### PR DESCRIPTION
**Pull request by aarthykc**
To check if tokens which have a space between them are global or generated per index. 

cc @yhahn, @arunasank

### CODE REVIEW

**Comment by aarthykc** [comment]
Added tests to the validator, throws an error if it uses global tokens. 

Would be great if you could take a look at this, @yhahn? 

cc @arunasank

File: index.js [hunk 1]
```diff
@@ -231,6 +236,12 @@ function boundsIntersect(a, b) {
     return true;
 }
 
+function tokenValidator(token_replacer) {
+    if (token_replacer[0].from.toString().indexOf(' ') >= 0 || token_replacer[0].to.toString().indexOf(' ') >= 0) {
+        return true;
+    }
```
Comment (yhahn) [inline index.js]: Hm, this looks like it's checking the very first `token_replacer[0]` from/to pair. Do you want to see if we could check them all?

(You can add a test for this by having an index with multiple geocoder_token pairs and having one of the invalid ones be later in the list).

**Comment by yhahn** [comment]
@aarthykc want to update the name of the test? And then I think this is ready for :ship:!

**Comment by mattficke** [comment]
@aarthykc @yhahn Hm, is there a way to accomplish this by actually checking if there's a cardinality change, rather than just checking if there are spaces? This current version means that all CJK tokens must be global (becuase of unidecode), and makes it more difficult to test.

Example that doesn't work now: `geocoder_tokens: {"[３3]丁目": "三丁目"}`

Thinking something like this instead:
```js
if (token_replacer[i].from.toString().split(' ').length !== token_replacer[i].to.toString().split(' ').length
```

**Comment by mattficke** [comment]
Turns out unidecode makes this somewhat more complicated since CJK queries are split per token. In my above example it turns into this, with lengths differing by 1:
```
from: /(\W|^)[33]Ding Mu (\W|$)/gi,
to: '$1San Ding Mu $2'
```
Need to mull it over more. One possibility is that this error shouldn't throw in Carmen, it should be in either api-geocoder or mapbox-places, since those are where the tokens actually live.

**Comment by mattficke** [comment]
@aarthy To clarify what's happening here, `3丁目` (3-chome, in latin characters) is an address component that can be written with either western or Japanese numerals. `三丁目` is the equivalent with all Japanese characters.

Our data source uses Japanese numerals, so I want to add an index-level token replacer to match queries with western numerals.

CJK queries are split per character, so the query is `3,丁,目`, and then the `from` token is unidecoded during the replacer creation process.

Here's the unidecoded versions of both (note the space after `Mu`, which unidecode typically inserts):

* `3丁目` --> `3Ding Mu `
* `三丁目` --> `San Ding Mu `

**Pull request merged**